dev/release/verify-release-candidate.sh

```diff
@@ -788,7 +788,7 @@ test_python() {
   pushd python
 
   # Build pyarrow
-  python setup.py build_ext --inplace
+  python -m pip install .
```

Comment:
```suggestion
  python -m pip install -e .
```

(if we want to keep the rest of the code below as is, it needs to be an inplace install)